src/main/scala/mesosphere/marathon/state/Container.scala

```diff
@@ -47,15 +47,19 @@ object Container {
         name: Option[String] = None,
         labels: Map[String, String] = Map.empty[String, String]) {
 
-      require(protocol == "tcp" || protocol == "udp", "protocol can only be 'tcp' or 'udp'")
+      require(
+        protocol == "tcp" || protocol == "udp" || protocol == "both",
+        "protocol can only be 'tcp', 'udp' or 'both'"
+      )
     }
 
     object PortMapping {
       val TCP = "tcp"
       val UDP = "udp"
+      val BOTH = "both"
```

Comment:
Minor: do not like the protocol name `both` too much but don't have a better one. 
Other options `tcp+udp` (but this is commutative)?